activemerchant.gemspec

```diff
@@ -26,7 +26,7 @@ Gem::Specification.new do |s|
   s.add_dependency('builder', '>= 2.1.2', '< 4.0.0')
   s.add_dependency('i18n', '>= 0.6.9')
   s.add_dependency('nokogiri', '~> 1.4')
-  s.add_dependency('rexml', '~> 3.2.5')
+  s.add_dependency('rexml', '~> 3.3.2')
```

Comment:
I'm wondering why it has to be so tight? This will still prevent minor version upgrades, which generally _shouldn't_ cause breakages. Would the following be suitable to pin to major version 3 while staging above the version required by the CVE?

```suggestion
  s.add_dependency('rexml', '~> 3.3', '>=3.3.2')
```

Reply:
Thanks updated.

Reply:
That solves the immediate problem, but doesn't prevent this scenario from happening again.

To help avoid this type of problem in the future, and to better align with the other `add_dependency` lines in the gemspec, I think we would prefer this:

```
s.add_dependency('rexml', '>= 3.3.2')
```

In other words, I don't think we have any reason for a pessimistic version constraint (meaning `~>`) on `rexml`, because there doesn't seem to be any reason to believe future versions would break `active_merchant`.

For background, the original pessimistic constraint was added in eed45fe28a355626dfb0f560ede5cf719da4e9f9 simply. There is no documented reason for a pessimistic constraint, so I think the version constraint was overly tight from the beginning. The [gemspec declarations](https://github.com/activemerchant/active_merchant/blame/master/activemerchant.gemspec#L25) for `activesupport`, `builder`, and `i18n` do not have pessimistic constraints, which seems correct.

Reply:
@dan-jensen Assuming that `rexml` is following semantic versioning correctly, wouldn't we want a conscious, verified step to any 4.0 version to check for breaking changes, however remote they may be?  As such I think the suggestion from @ozydingo sets a sensible base and ceiling.

Reply:
@damienjbyrne you're right, it's definitely sensible. But there is a tradeoff. We can choose to prioritize breaking changes or vulnerabilities. Breaking changes tend to be obvious, and easy to resolve (simply don't change). Vulnerabilities tend to be non-obvious, and harder to resolve (because change is required). I think it makes sense to prioritize vulnerabilities, and flexibility in general, like the majority of this gemspec already does. But I didn't intend to cause a delay here. If there's doubt about flexible specifications, I support merging this PR now.